src/utils/check-page-for-error.js

```diff
@@ -0,0 +1,34 @@
+import statusCodes, { OK } from '../constants/status-codes.js';
+
+/**
+ * Checks if a page has a required radio answer before sending an event to the grant type machine
+ * @param {object} grantTypeMachineService - The grant type machine service
+ * @param {object} request - The request object
+ * @param {object} h - The response toolkit
+ * @returns {object} The response object
+ */
+export const checkPageForError = (grantTypeMachineService, request, h) => {
```

Comment:
This solution is three times as long as the current one. I think the if/else style used is clear and readable.

Reply:
In my honest opinion the original solution is more concise but harder to extend. Adding validation for a new question type requires modifying the same function, risking errors and reduced clarity. The additional layers (validation functions and map) introduce some complexity but keep the logic centralised and modular.

It has significant advantages:

- abstraction of validation and logic (eparates validation logic by question type (radio, text, etc.) into dedicated functions (validateRadioAnswer, validateTextAnswer),
- enables easy addition of new question types. Adding validation for a new type requires implementing a function and adding it to the map, without modifying the main validation function,
- each validation function handles only one question type, adhering to the Single Responsibility Principle,
- includes a detailed JSDoc comment and function names that clearly describe their purpose, improving readability and maintainability.

Thing is the list of question types is expected to grow and more modular would be to avoid these if elses and have this solution instead.

Reply:
This has been resolved now by moving the validation logic inside state machine using guards and setting errors via actions.